### PR TITLE
Add nidigital system tests for remaining source and capture methods

### PR DESCRIPTION
**Pull request by ni-jfitzger**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [x] ~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~

- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

* Adds test `test_create_source_waveform_serial`
* Adds test `test_create_source_waveform_from_file_tdms`
* Adds test `test_write_source_waveform_data_from_file_tdms`
* Adds test `test_create_capture_waveform_serial`
* Adds test `test_create_capture_waveform_from_file_digicapture`

### What testing has been done?

Ran system tests on a system with ni-digital 19.0.1 installed.

### CODE REVIEW

**Comment by ni-jfitzger** [comment]
@sbethur Please review this PR very carefully.  The tests pass, but I really have no idea what I'm doing or how these methods really work.  I just stole code from other tests and messed with it until it passed.  There's a fair chance that I missed, messed up or overlooked something that matters outside the execution of the methods being tested.

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 1]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
...
+        waveform_name='src_wfm',
+        waveform_file_path=get_test_file_path(test_name, 'source_waveform.tdms'))
+
+    # burst the waveform to confirm that configuration and loading went okay
+    pass_fail = multi_instrument_session.burst_pattern(start_label='new_pattern')
+    assert pass_fail == {0: True, 1: True}
+
+
+def test_create_capture_waveform_serial(multi_instrument_session):
+    test_name = 'test_fetch_capture_waveform'
```
Comment (sbethur) [inline src/nidigital/system_tests/test_system_nidigital.py:1025]: Rename'test_fetch_capture_waveform' to 'test_fetch_capture_waveform_parallel' and create a new project - 'test_fetch_capture_waveform_serial' with the same souce configuration as that in test_create_source_waveform_serial.

Use this pattern:
image

Capture configuration: LSB, 2 bits, and HI0.

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 2]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
...
+    configure_session(multi_instrument_session, test_name)
+    multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+    num_samples = 256
```
Comment (sbethur) [inline src/nidigital/system_tests/test_system_nidigital.py:1028]: = 2

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 3]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
...
+
+    # The pattern references a wfm 'src_wfm', so we have to load it before we can burst
+    multi_instrument_session.pins['LO0'].create_source_waveform_serial(
+        waveform_name='src_wfm',
+        data_mapping=nidigital.SourceDataMapping.BROADCAST,
+        sample_width=2,
+        bit_order=nidigital.BitOrder.LSB)
+    multi_instrument_session.write_source_waveform_broadcast(
+        waveform_name='src_wfm',
+        waveform_data=[0])
```
Comment (sbethur) [inline src/nidigital/system_tests/test_system_nidigital.py:1043]: waveform_data=[1, 2]

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 4]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
...
+    multi_instrument_session.burst_pattern(start_label='new_pattern')
+
+    # Fetch to confirm that configuration went okay
+    multi_instrument_session.sites[1, 0].fetch_capture_waveform(
```
Comment (sbethur) [inline src/nidigital/system_tests/test_system_nidigital.py:1047]: Assert thse after fetch():
```
    assert sorted(fetched_waveforms.keys()) == sorted([0, 1])
    assert all(len(fetched_waveforms[site]) == num_samples for site in fetched_waveforms)
```

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 5]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
...
+        waveform_name='capt_wfm',
+        samples_to_read=num_samples)
+
+
+def test_create_capture_waveform_from_file_digicapture(multi_instrument_session):
```
Comment (sbethur) [inline src/nidigital/system_tests/test_system_nidigital.py:1052]: Almost all comments for `test_create_capture_waveform_serial` apply for this test too.

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 6]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
     assert result == {0: True, 3: True}
 
 
+def test_create_source_waveform_serial(multi_instrument_session):
+    test_name = 'test_source_waveform_parallel_broadcast'
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py]: Done.

File: src/nidigital/system_tests/test_system_nidigital.py [hunk 7]
```diff
@@ -961,6 +961,121 @@ def test_configure_pattern_burst_sites(multi_instrument_session):
...
+    multi_instrument_session.pins['LO0'].create_source_waveform_serial(
+        waveform_name='src_wfm',
+        data_mapping=nidigital.SourceDataMapping.BROADCAST,
+        sample_width=2,
+        bit_order=nidigital.BitOrder.LSB)
+
+    # load and burst the waveform to confirm that configuration went okay
+    multi_instrument_session.write_source_waveform_broadcast(
+        waveform_name='src_wfm',
+        waveform_data=[i for i in range(4)])
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py]: Done.

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 1 above)
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py]: Done, though I named the new project 'test_create_source_waveform_serial'.
I updated the test to use your recommended capture configuration, in addition to creating that configuration within the new project.

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 2 above)
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py]: Updated.

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 3 above)
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py]: Updated

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 4 above)
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py]: Done.

File: src/nidigital/system_tests/test_system_nidigital.py (hunk 5 above)
Comment (ni-jfitzger) [reply src/nidigital/system_tests/test_system_nidigital.py:1117]: Updated.
I decided to point this test at the project used by `test_create_capture_waveform_serial` and load the source from the .tdms file.

**Comment by ni-jfitzger** [comment]
Note that between the renamed test_files project and the new test_create_capture_waveform_serial project, some files may be exactly the same, leading Git to be confused about which file was renamed and which was added.

**Comment by sbethur** [comment]
test this please

**Pull request merged**